archspec/cpu/microarchitecture.py

```diff
@@ -300,6 +299,18 @@ def tuplify(ver):
         msg = msg.format(self.name, compiler, version)
         raise UnsupportedMicroarchitecture(msg)
 
+class AARCH64_Microarchitecture(Microarchitecture):
+    """Represents a specific AARCH64 CPU micro-architecture.
+
+    The only addition to the Microarchitecture class it inherits
+    from is the cpuid field that for specific aarch64 architectures
+    should be present in the definition of the micro-architecture.
+    """
+
+    def __init__(self, name, parents, vendor, features, compilers, generation=0, cpuid = None):
+        self.cpuid = cpuid
+        super().__init__(name, parents, vendor, features, compilers, generation)
+
```

Comment:
Let's not create a separate class, and just add a new field to `Microarchitecture`. We already have `generation` which is relevant only on `ppc`.

Reply:
Agreed and removed.